rescript/tests/test_derep.py

```diff
@@ -349,3 +436,20 @@ def _backfill_series(series, rank_handles=default_rank_handle):
         exp_taxa = trimmed_taxa.apply(lambda x: x + ';n;u;t;s')
         backfilled_taxa = _backfill_series(trimmed_taxa, custom_rank_handles)
         pdt.assert_series_equal(backfilled_taxa, exp_taxa, check_names=False)
+
+        # alt  / longer ranks
+        # same as fungal data but then adding subspecies prefixes
+        alt_rank_handles = ('d__; sk__; k__; ks__; p__; ps__; c__; o__; sf__;'
+                            ' g__; s__; ssb__; for__')
+        f_taxa = self.fungi_taxa.view(pd.Series).sort_index()
+        f_taxa = f_taxa.apply(lambda x: ';'.join(
+                          _return_stripped_taxon_rank_list(x)))
+        backfilled_f_taxa = _backfill_series(f_taxa, alt_rank_handles)
+        trimmed_f_taxa = backfilled_f_taxa.apply(
+            lambda x: ';'.join(x.split(';')[:9]))
+        # manually backfill
+        exp_f_taxa = trimmed_f_taxa.apply(lambda x: x + ';g__;s__;ssb__;for__')
+        updated_backfilled_f_taxa = _backfill_series(trimmed_f_taxa,
```

Comment:
hey @mikerobeson this new test is a bit confusing. I am not following the logic... you backfill, trim, then backfill again. And the expected taxonomy is builf off of the taxonomy that was backfilled once. Could you do the following to make this test a bit more transparent?
1. Move to its own unit test so that any failures are more transparently reported.
2. explicitly write out the expected taxonomy instead of building this off of backfilled data.
3. there are quite a few extra unneeded lines here because you test on a series. You could instead just test on a list of, say, 3 taxonomy strings of different lengths and contents (or however many edge cases you see) and this would eliminate the need for several lines of code.

Reply:
HI @nbokulich, looking at this again with fresh eye... yeah... not sure what I was thinking. 🤦‍♂️ 

Hopefully, this new refactored version is in line with what you suggested.